src/middlewared/middlewared/plugins/enclosure_/jbof_enclosures.py

```diff
@@ -101,8 +101,23 @@ def map_es24n(model, rclient, uri):
     }
     mapped = dict()
     for disk in all_disks['Members']:
-        slot = int(disk['Id'])
-        if found := mounted_disks.get(disk['SerialNumber']):
+        if disk['State'] == 'Absent':
+            mapped[slot] = None
+            continue
+
+        sn = disk.get('SerialNumber')
+        if not sn:
+            mapped[slot] = None
+            continue
+
+        slot = disk.get('Id', '')
```

Comment:
Should this be the first thing to be done in the loop body? We use `slot` above and it might be undefined

Reply:
Juggling too many things. Good catch, I've fixed.